src/ripple/protocol/LedgerFormats.h

```diff
@@ -24,76 +24,177 @@
 
 namespace ripple {
 
-/** Ledger entry types.
+/** Identifiers for on-ledger objects.
 
-    These are stored in serialized data.
+    Each ledger object requires a unique type identifier, which is stored
+    within the object itself; this makes it possible to iterate the entire
+    ledger and determine each object's type and verify that the object you
+    retrieved from a given hash matches the expected type.
 
-    @note Changing these values results in a hard fork.
+    @warning The range of valid on-ledger identifiers is between 1 and 32767
```

Comment:
How about adding a parenthetical "(all positive integers representable with 15 bits)"?

Reply:
I think that the existing comment is confusing enough and (imo) adding that increase confusion. How about using _2^15-1_ instead, with an additional reference to the `minLedgerEntryType` and `maxLedgerEntryType` constants, described below?